light_controller/ir_toy/ir_toy.py

```diff
@@ -0,0 +1,351 @@
+        free in the USB buffer, currently always 62. The software interfacing the IR Toy should wait for each request
+        before sending the data.
+        :return: None
+        """
+        self.writeByte(CommandEnum.enableTX_Handshake)
+
+    def reset(self):
+        """
+        Reset the device
+        :return: None
+        """
+        self.writeArray(self.TERMINATOR)
+        self.writeArray((CommandEnum.reset,) * 5)
+        time.sleep(0.05)
+
+    def getVersion(self):
+        """
+        Get the hardware and firmware versions
+        :return: hardware string, firmware version
+        """
+        try:
+            self.reset()
+            self.writeByte(ord('v'))
+            buffer = self.read(4)
+            hardware, revision = struct.unpack('>2s2s', buffer)
+            return hardware, int(revision)
+        finally:
+            self.setSamplingMode()
+
+    def setTransmitMode(self):
+        """
+        Setup the device for transmitting.
+        Enable handshakes and reports
+        :return: TX buffer size
+        """
+        self.enableTX_Handshake()
+        self.enableTX_ByteCount()
+        self.enableTX_NotifyOnComplete()
+        bufferSize = self.enableTransmitMode()
+        return bufferSize
+
+    def setSamplingMode(self):
+        """
+        Setup the device for sampling
+        :return: None
+        """
+        self.reset()
+        self.writeByte(ord('S'))
+        self.protocolVersion = self.read(3)
+
+    def receiveSignal(self):
+        """
+        Record a signal
+
+        :return: List of bytes representing the ir code.
+        """
+        self.setSamplingMode()
+        bytesToRead = 1
+        readCount = 0
+        irCode = []
+
+        while (True):
+            readByte = self.read(bytesToRead)
+            if not readByte:
+                continue
+            readByte = ord(readByte)
+            irCode.append(readByte)
+            if irCode[-2:] == self.TERMINATOR:
+                break
+            readCount += 1
+        return irCode
+
+    def transmitCode(self, code):
+        """
+        Transmit an IR code
+        :param code: A string of bytes to send in the same format as the recording
+        :return: Success
+        """
+        if len(code) < 2 or len(code) % 2:
+            raise ValueError("Length of code must be a multiple of 2")
+
+        if code[-2:] != self.TERMINATOR:
+            code.extend(self.TERMINATOR)
+
+        idx = 0
+        bytesSent = 0
+        bufferSize = self.setTransmitMode()
+        try:
+            while bytesSent < len(code):
+                sent = self.writeArray(code[idx:idx + bufferSize])
+                idx += sent
+                bytesSent += sent
+                bufferSize = ord(self.read(1))
+
+            # Wait for IR Toy to work out we're finished.
+            time.sleep(0.05)
+            completionReport = self.read(4)
```

Comment:
[minor] just a suggestion. We can have separate methods, like:

```
def read_next_buffer_size(self):
    return self.read(1)

def read_completion_report(self):
    return self.read(4)
```

This way we can avoid confusion with the numbers.